### DISCUSSION TITLE
Unneccessary Group BY Clause added to query; Missing Order by Clause

### DISCUSSION THREAD

**Discussion by kevin-struckhoff** [apache/superset#18319] [Q&A / Help]
Superset is adding a Group by clause to the SQL query where none is specified in the line chart setup screen. It also does not include the sort by column which was specified, which may be a separate issue altogether.

Since the dataset does not require grouping by date, there is no need for a group by clause. 

#### How to reproduce the bug

1. Create a Line Chart
2. Select Time/Date Column and Needed Metrics & Filters
3. Do Not select a Group By column
4. Set a Sort By Clause 
5. Hit the Run Button
6. Receive the runtime error shown in the screenshot.
7. Review SQL query which shows a Group By clause but not an Order By clause as shown in the screenshot.

If I select the metrics which are summed, for example sum(num_steps), then set the Group By column, it will run, but it also adds an extra _timestamp column to the query (which is not even my dataset to begin with) and doesn't chart correctly since 2 date/time columns are listed in the query instead of just the one I specified.

![line-chart-GroupBy-error](https://user-images.githubusercontent.com/88514509/128614770-1c6c6594-fa92-4ed5-89db-93989c00e04d.png)

### Environment

- superset version: `1.2` on CentOS 8
- python version: `3.8` on CentOS 8
- node.js version: none
- any feature flags active:

**Comment by zhaoyongjie** [comment]
Hi @kevin-struckhoff. The Line Chart is a time-series chart, this means that there is always a time column in the query, in the other words, the data always is aggregated by time-column and other selected columns (group by control in this chart)

and you need to define how to aggregate for column in the Metrics, your SQL seems that did not define aggregate for columns.

**Comment by kamalkeshavani-aiinside** [comment]
And SORT BY option is used only to decide the top N series values, when you add a GROUP BY column and set SERIES LIMIT.
Ex., You group by using a column with 10 distinct values, and set series limit to 5. Then Sort By will decide which 5 values will be seen in the chart.
